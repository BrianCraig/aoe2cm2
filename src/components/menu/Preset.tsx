import * as React from "react";
import {Link} from "react-router-dom";

class Preset extends React.Component<object, object> {
    public render() {
        return (
            <div>
                <p>Description of a single preset</p>
                <Link to='/'>Go to index</Link>
            </div>
        );
    }
}

export default Preset;