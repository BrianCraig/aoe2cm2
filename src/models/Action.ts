enum Action {
    PICK = 'PICK',
    EXCLUSIVE_PICK = 'EXCLUSIVE_PICK',
    GLOBAL_PICK = 'GLOBAL_PICK',
    HIDDEN_PICK = 'HIDDEN_PICK',
    HIDDEN_EXCLUSIVE_PICK = 'HIDDEN_EXCLUSIVE_PICK',
    BAN = 'BAN',
    EXCLUSIVE_BAN = 'EXCLUSIVE_BAN',
    GLOBAL_BAN = 'GLOBAL_BAN',
    HIDDEN_BAN = 'HIDDEN_BAN',
    HIDDEN_EXCLUSIVE_BAN = 'HIDDEN_EXCLUSIVE_BAN',
    HIDDEN_GLOBAL_BAN = 'HIDDEN_GLOBAL_BAN',
    SNIPE = 'SNIPE',
    HIDDEN_SNIPE = 'HIDDEN_SNIPE',
    
    // REVEAL_PICKS = 'REVEAL_PICKS',
    // REVEAL_BANS = 'REVEAL_BANS',
    // REVEAL_SNIPES = 'REVEAL_SNIPES',
    REVEAL_ALL = 'REVEAL_ALL'
}

export default Action;