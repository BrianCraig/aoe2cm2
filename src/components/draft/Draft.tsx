import * as React from "react";
import CivGrid from "./CivGrid";
import Messages from "../../containers/Messages";
import DraftState from "./DraftState";
import TurnRow from "./TurnRow";
import Player from "../../constants/Player";
import Preset from "../../models/Preset";
import "../../types/DraftEvent";
import {DraftEvent} from "../../types/DraftEvent";
import {IDraftConfig} from "../../types/IDraftConfig";
import {WithTranslation, withTranslation} from "react-i18next";
import Modal from "../../containers/Modal";
import NameGenerator from "../../util/NameGenerator";
import {Link} from "react-router-dom";
import RoleModal from "../../containers/RoleModal";
import DraftIdInfo from "../../containers/DraftIdInfo";

interface IProps extends WithTranslation {
    nameHost: string;
    nameGuest: string;
    hostConnected: boolean;
    guestConnected: boolean;
    whoAmI?: Player;
    ownName: string | null;
    preset: Preset;
    nextAction: number;

    onActionCompleted?: (message: DraftEvent) => void;
    onDraftConfig?: (message: IDraftConfig) => void;
    onNextAction?: () => void;
    onSetNameHostAction?: (name: string) => void;
    onSetNameGuestAction?: (name: string) => void;
    triggerConnect: () => void;
    triggerSetRole: (name: string, role: Player) => void;
    triggerDisconnect?: () => void;
    showRoleModal: () => void;
    showNameModal: () => void;
    setOwnRole: (role: Player) => void;
}

interface IState {
    joined: boolean;
}

class Draft extends React.Component<IProps, IState> {

    state = {joined: false};

    componentDidMount(): void {
        this.props.triggerConnect();
    }

    componentDidUpdate(prevProps: Readonly<IProps>, prevState: Readonly<IState>, snapshot?: any): void {
        console.log('cdU', this.props);
        if (this.props.whoAmI === undefined) {
            console.log('is undefined');
            if (!this.props.hostConnected || !this.props.guestConnected) {
                this.props.showRoleModal();
            } else {
                this.props.setOwnRole(Player.NONE);
                this.setState({joined: true});
            }
        } else if (this.props.whoAmI !== Player.NONE && !this.state.joined) {
            console.log('is other');
            let username: string | null = NameGenerator.getNameFromLocalStorage(this.props.ownName);
            console.log("componentDidMount", this.props.triggerSetRole, username);
            if (username !== null) {
                console.log('triggering JOIN');
                this.props.triggerSetRole(username, this.props.whoAmI);
                this.setState({joined: true});
            } else {
                this.props.showNameModal();
            }
        }
    }

    public render() {
        const presetName: string = this.props.preset.name;
        const turns = this.props.preset.turns;


        return (
            <div id="container">
                <div style={{position: 'absolute', top: '8px', left: '8px'}}>
                    <span onClick={this.props.triggerDisconnect}><Link to="/"><span
                        className="back-icon header-navigation">back</span></Link></span>
                </div>

                <Modal inDraft={true}/>
                <RoleModal/>

                <div className="draft-content">

                    <div id="draft-title" className="centered text-primary info-card">{presetName}</div>

                    <TurnRow turns={turns}/>

                    <DraftState nameHost={this.props.nameHost} nameGuest={this.props.nameGuest}
                                preset={this.props.preset}/>

                    <div>
                        <div id="action-text" className="centered">
                            <div className="action-string info-card text-primary">
                                <Messages/>
                            </div>
                        </div>
                    </div>

                    <DraftIdInfo/>

                    <CivGrid civilisations={this.props.preset.civilisations}/>

                </div>
            </div>
        );
    }
}

export default withTranslation()(Draft);