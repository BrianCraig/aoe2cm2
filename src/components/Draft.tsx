import * as React from "react";
import CivGrid from "./CivGrid";
import Messages from "../containers/Messages";
import Players from "./Players";
import TurnRow from "./TurnRow";
import Player from "../models/Player";
import Preset from "../models/Preset";
import "../models/DraftEvent";
import {DraftEvent} from "../models/DraftEvent";
import {IDraftConfig} from "../models/IDraftConfig";
import {WithTranslation, withTranslation} from "react-i18next";
import Modal from "../containers/Modal";
import NameGenerator from "../models/NameGenerator";
import {Link} from "react-router-dom";

interface IProps extends WithTranslation {
    nameHost: string;
    nameGuest: string;
    whoAmI: Player;
    ownName: string | null;
    preset: Preset;
    nextAction: number;

    onActionCompleted?: (message: DraftEvent) => void;
    onDraftConfig?: (message: IDraftConfig) => void;
    onNextAction?: () => void;
    onSetNameHostAction?: (name: string) => void;
    onSetNameGuestAction?: (name: string) => void;
    triggerJoin?: (name: string) => void;
    triggerDisconnect?: () => void;
    showNameModal?: () => void;
}

interface IState {
    nextStep: number;
}

class Draft extends React.Component<IProps, IState> {

    componentDidMount(): void {
        let username: string | null = NameGenerator.getNameFromLocalStorage(this.props.ownName);
        console.log("componentDidMount", this.props.triggerJoin, username);
        if (this.props.triggerJoin !== undefined && username !== null) {
            console.log('triggering JOIN');
            this.props.triggerJoin(username);
        } else if (this.props.showNameModal !== undefined && username === null) {
            this.props.showNameModal();
        }
    }

    public render() {
        const presetName: string = this.props.preset.name;
        const turns = this.props.preset.turns;


        return (
            <div id="container">
                <div style={{position: 'absolute', top: '8px', left: '8px'}}>
                    <span onClick={this.props.triggerDisconnect}><Link to="/"><span className="back-icon header-navigation">back</span></Link></span>
                </div>

                <Modal inDraft={true}/>

                <div className="draft-content">

                    <div id="draft-title" className="centered text-primary info-card">{presetName}</div>

                    <TurnRow turns={turns}/>

                    <Players nameHost={this.props.nameHost} nameGuest={this.props.nameGuest} preset={this.props.preset}/>

                    <div>
                        <div id="action-text" className="centered">
                            <div className="action-string info-card text-primary">
                                <Messages/>
                            </div>
                        </div>
                    </div>

                    <CivGrid civilisations={this.props.preset.civilisations}/>

                </div>
            </div>
        );
    }
}

export default withTranslation()(Draft);