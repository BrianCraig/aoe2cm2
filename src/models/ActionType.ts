import Action from "./Action";

enum ActionType {
    PICK = "pick",
    BAN = "ban",
    SNIPE = "snipe",
    REVEAL = "reveal",
    NOTHING = "nothing"
}

export function actionTypeFromAction(action: Action) {
    switch (action) {
        case Action.PICK:
        case Action.EXCLUSIVE_PICK:
        case Action.GLOBAL_PICK:
        case Action.HIDDEN_PICK:
        case Action.HIDDEN_EXCLUSIVE_PICK:
            return ActionType.PICK;
        case Action.BAN:
        case Action.EXCLUSIVE_BAN:
        case Action.GLOBAL_BAN:
        case Action.HIDDEN_BAN:
        case Action.HIDDEN_EXCLUSIVE_BAN:
        case Action.HIDDEN_GLOBAL_BAN:
            return ActionType.BAN;
        case Action.SNIPE:
        case Action.HIDDEN_SNIPE:
            return ActionType.SNIPE;
        // case Action.REVEAL_PICKS:
        // case Action.REVEAL_BANS:
        // case Action.REVEAL_SNIPES:
        case Action.REVEAL_ALL:
            return ActionType.REVEAL;
    }
}

export default ActionType;