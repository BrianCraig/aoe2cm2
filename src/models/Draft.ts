import Preset from "./Preset";
import {IStoreState} from "../types";
import Player from "./Player";
import {DraftEvent} from "./DraftEvent";
import NameGenerator from "./NameGenerator";
import Turn from "./Turn";
import Civilisation from "./Civilisation";
import Action from "./Action";
import PlayerEvent from "./PlayerEvent";
import {Util} from "./Util";

class Draft implements IStoreState {
    public nameHost: string;
    public nameGuest: string;
    public hostReady: boolean;
    public guestReady: boolean;
    public whoAmI: Player;
    public ownName: string | null = null;
    public readonly preset: Preset;
    public nextAction: number = 0;
    public events: DraftEvent[] = [];
    public showModal: boolean;


    public language: string = 'en-GB';

    constructor(nameHost: string, nameGuest: string, preset: Preset) {
        this.nameHost = nameHost;
        this.nameGuest = nameGuest;
        this.preset = preset;
        this.whoAmI = Player.HOST;
        this.hostReady = false;
        this.guestReady = false;
        this.ownName = NameGenerator.getNameFromLocalStorage();
        this.showModal = this.ownName === null;
    }

    public static from(source: Draft): Draft {
        const draft: Draft = new Draft(source.nameHost, source.nameGuest, source.preset)
        draft.hostReady = source.hostReady;
        draft.guestReady = source.guestReady;
        draft.whoAmI = source.whoAmI;
        draft.ownName = source.ownName;
        draft.nextAction = source.nextAction;
        draft.events = source.events;
        draft.showModal = source.showModal;
        return draft;
    }

    public setWhoAmI(whoAmI: Player) {
        this.whoAmI = whoAmI;
    }

    public static playersAreReady(draft: IStoreState) {
        return draft.hostReady && draft.guestReady;
    }

    public hasNextAction(): boolean {
        return this.events.length < this.preset.turns.length;
    }

    public draftCanBeStarted(): boolean {
        return this.hostReady && this.guestReady;
    }

    public getExpectedAction(): Turn | null {
        if (!Draft.playersAreReady(this)) {
            return null;
        }
        if (this.hasNextAction()) {
            return this.preset.turns[this.events.length];
        }
        return null;
    }

    public getGlobalBans(): Civilisation[] {
        const globalBanTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.action === Action.GLOBAL_BAN
            });
        const globalBans: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (globalBanTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                globalBans.push(playerEvent.civilisation);
            }
        });
        return globalBans;
    }

    public getBansForPlayer(player: Player): Civilisation[] {
        if (player === Player.NONE) {
            return [];
        }
        const opponent = player === Player.HOST ? Player.GUEST : Player.HOST;
        const opponentBanTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.player === opponent && Util.isNonglobalBan(turn.action);
            });

        const opponentBans: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (opponentBanTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                opponentBans.push(playerEvent.civilisation);
            }
        });
        return opponentBans;
    }

    public getExclusivePicks(player: Player): Civilisation[] {
        const exclusivePickTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.player === player && turn.action === Action.PICK;
            });

        const exclusivePicks: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (exclusivePickTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                exclusivePicks.push(playerEvent.civilisation);
            }
        });
        return exclusivePicks;
    }

    public getGlobalPicks(): Civilisation[] {
        const globalPickTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.action === Action.GLOBAL_PICK;
            });

        const globalPicks: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (globalPickTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                globalPicks.push(playerEvent.civilisation);
            }
        });
        return globalPicks;
    }

    public getExclusiveBansByPlayer(player: Player): Civilisation[] {
        if (player === Player.NONE) {
            return [];
        }
        const exclusiveBanTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.player === player && turn.action === Action.BAN;
            });

        const exclusiveBans: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (exclusiveBanTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                exclusiveBans.push(playerEvent.civilisation);
            }
        });
        return exclusiveBans;
    }

    public getPicks(player: Player): Civilisation[] {
        const pickTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.player === player && Util.isPick(turn.action);
            });

        const picks: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (pickTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                picks.push(playerEvent.civilisation);
            }
        });
        return picks;
    }

    getSnipes(player: Player): Civilisation[] {
        const snipeTurns: boolean[] = this.preset.turns
            .map((turn): boolean => {
                return turn.player === player && Util.isSnipe(turn.action);
            });

        const snipes: Civilisation[] = [];
        this.events.forEach((draftEvent: DraftEvent, index: number) => {
            if (snipeTurns[index]) {
                const playerEvent = draftEvent as PlayerEvent;
                snipes.push(playerEvent.civilisation);
            }
        });
        return snipes;
    }


}

export default Draft