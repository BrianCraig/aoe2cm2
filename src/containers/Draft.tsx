import Draft from '../components/Draft';
import * as actions from '../actions/';
import {IStoreState} from '../types';
import {connect} from 'react-redux';
import {Dispatch} from 'redux';
import Player from "../models/Player";

export function mapStateToProps({nameHost, nameGuest, whoAmI, preset, nextAction, events}: IStoreState) {
    return {
        events,
        nameGuest,
        nameHost,
        nextAction,
        preset,
        whoAmI
    }
}

export function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
    return {
        onNextAction: () => dispatch(actions.completeAction()),
        onSetNameGuestAction: () => dispatch(actions.setName(Player.GUEST, "Eumel Guest")),
        onSetNameHostAction: () => dispatch(actions.setName(Player.HOST, "Eumel Host")),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Draft);