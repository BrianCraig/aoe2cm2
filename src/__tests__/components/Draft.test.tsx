import {shallow} from "enzyme";
import * as React from "react";
import Draft from "../../components/Draft";
import {default as ModelDraft} from "../../models/Draft";
import Preset from "../../models/Preset";

it('Draft renders correctly', () => {
    const draft:ModelDraft= new ModelDraft('Sneaky Saladin', 'Beastly Barbarossa', Preset.SAMPLE);
    const component = shallow(<Draft config={draft}/>);
    expect(component).toMatchSnapshot();
});
