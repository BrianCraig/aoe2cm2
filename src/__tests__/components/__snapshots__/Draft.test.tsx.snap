// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draft renders correctly 1`] = `
<div
  className="draft-content"
>
  <div
    className="centered text-primary info-card"
    id="draft-title"
  >
    Default Preset
  </div>
  <TurnRow
    turns={
      Array [
        Turn {
          "action": "GLOBAL_BAN",
          "player": "HOST",
        },
        Turn {
          "action": "GLOBAL_BAN",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_BAN",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_BAN",
          "player": "GUEST",
        },
        Turn {
          "action": "REVEAL_BANS",
          "player": "NONE",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_PICK",
          "player": "GUEST",
        },
        Turn {
          "action": "REVEAL_PICKS",
          "player": "NONE",
        },
        Turn {
          "action": "HIDDEN_SNIPE",
          "player": "HOST",
        },
        Turn {
          "action": "HIDDEN_SNIPE",
          "player": "GUEST",
        },
        Turn {
          "action": "REVEAL_ALL",
          "player": "NONE",
        },
      ]
    }
  />
  <Players
    nameGuest="Beastly Barbarossa"
    nameHost="Sneaky Saladin"
    preset={
      Preset {
        "civilisations": Array [
          Civilisation {
            "gameVersion": 1,
            "name": "Britons",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Byzantines",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Celts",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Chinese",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Franks",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Goths",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Japanese",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Mongols",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Persians",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Saracens",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Teutons",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Turks",
          },
          Civilisation {
            "gameVersion": 1,
            "name": "Vikings",
          },
          Civilisation {
            "gameVersion": 2,
            "name": "Aztecs",
          },
          Civilisation {
            "gameVersion": 2,
            "name": "Huns",
          },
          Civilisation {
            "gameVersion": 2,
            "name": "Koreans",
          },
          Civilisation {
            "gameVersion": 2,
            "name": "Mayans",
          },
          Civilisation {
            "gameVersion": 2,
            "name": "Spanish",
          },
          Civilisation {
            "gameVersion": 3,
            "name": "Italians",
          },
          Civilisation {
            "gameVersion": 3,
            "name": "Incas",
          },
          Civilisation {
            "gameVersion": 3,
            "name": "Indians",
          },
          Civilisation {
            "gameVersion": 3,
            "name": "Magyars",
          },
          Civilisation {
            "gameVersion": 3,
            "name": "Slavs",
          },
          Civilisation {
            "gameVersion": 4,
            "name": "Berbers",
          },
          Civilisation {
            "gameVersion": 4,
            "name": "Ethiopians",
          },
          Civilisation {
            "gameVersion": 4,
            "name": "Malians",
          },
          Civilisation {
            "gameVersion": 4,
            "name": "Portuguese",
          },
          Civilisation {
            "gameVersion": 5,
            "name": "Burmese",
          },
          Civilisation {
            "gameVersion": 5,
            "name": "Khmer",
          },
          Civilisation {
            "gameVersion": 5,
            "name": "Malay",
          },
          Civilisation {
            "gameVersion": 5,
            "name": "Vietnamese",
          },
        ],
        "name": "Default Preset",
        "turns": Array [
          Turn {
            "action": "GLOBAL_BAN",
            "player": "HOST",
          },
          Turn {
            "action": "GLOBAL_BAN",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_BAN",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_BAN",
            "player": "GUEST",
          },
          Turn {
            "action": "REVEAL_BANS",
            "player": "NONE",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_PICK",
            "player": "GUEST",
          },
          Turn {
            "action": "REVEAL_PICKS",
            "player": "NONE",
          },
          Turn {
            "action": "HIDDEN_SNIPE",
            "player": "HOST",
          },
          Turn {
            "action": "HIDDEN_SNIPE",
            "player": "GUEST",
          },
          Turn {
            "action": "REVEAL_ALL",
            "player": "NONE",
          },
        ],
      }
    }
  />
  <Connect(Messages) />
  <CivGrid
    civilisations={
      Array [
        Civilisation {
          "gameVersion": 1,
          "name": "Britons",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Byzantines",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Celts",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Chinese",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Franks",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Goths",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Japanese",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Mongols",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Persians",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Saracens",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Teutons",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Turks",
        },
        Civilisation {
          "gameVersion": 1,
          "name": "Vikings",
        },
        Civilisation {
          "gameVersion": 2,
          "name": "Aztecs",
        },
        Civilisation {
          "gameVersion": 2,
          "name": "Huns",
        },
        Civilisation {
          "gameVersion": 2,
          "name": "Koreans",
        },
        Civilisation {
          "gameVersion": 2,
          "name": "Mayans",
        },
        Civilisation {
          "gameVersion": 2,
          "name": "Spanish",
        },
        Civilisation {
          "gameVersion": 3,
          "name": "Italians",
        },
        Civilisation {
          "gameVersion": 3,
          "name": "Incas",
        },
        Civilisation {
          "gameVersion": 3,
          "name": "Indians",
        },
        Civilisation {
          "gameVersion": 3,
          "name": "Magyars",
        },
        Civilisation {
          "gameVersion": 3,
          "name": "Slavs",
        },
        Civilisation {
          "gameVersion": 4,
          "name": "Berbers",
        },
        Civilisation {
          "gameVersion": 4,
          "name": "Ethiopians",
        },
        Civilisation {
          "gameVersion": 4,
          "name": "Malians",
        },
        Civilisation {
          "gameVersion": 4,
          "name": "Portuguese",
        },
        Civilisation {
          "gameVersion": 5,
          "name": "Burmese",
        },
        Civilisation {
          "gameVersion": 5,
          "name": "Khmer",
        },
        Civilisation {
          "gameVersion": 5,
          "name": "Malay",
        },
        Civilisation {
          "gameVersion": 5,
          "name": "Vietnamese",
        },
      ]
    }
    socket={
      Socket {
        "_callbacks": Object {
          "$connect": Array [
            [Function],
          ],
          "$connecting": Array [
            [Function],
          ],
          "$playerEvent": Array [
            [Function],
          ],
          "$player_joined": Array [
            [Function],
          ],
        },
        "acks": Object {
          "0": [Function],
        },
        "connected": false,
        "disconnected": true,
        "flags": Object {},
        "ids": 1,
        "io": Manager {
          "_callbacks": Object {
            "$close": Array [
              [Function],
            ],
            "$open": Array [
              [Function],
            ],
            "$packet": Array [
              [Function],
            ],
          },
          "_randomizationFactor": 0.5,
          "_reconnection": true,
          "_reconnectionAttempts": Infinity,
          "_reconnectionDelay": 1000,
          "_reconnectionDelayMax": 5000,
          "_timeout": 20000,
          "autoConnect": true,
          "backoff": Backoff {
            "attempts": 0,
            "factor": 2,
            "jitter": 0.5,
            "max": 5000,
            "ms": 1000,
          },
          "connecting": Array [
            [Circular],
          ],
          "decoder": Decoder {
            "reconstructor": null,
          },
          "encoder": Encoder {},
          "encoding": false,
          "engine": Socket {
            "_callbacks": Object {
              "$error": Array [
                [Function],
              ],
              "$open": Array [
                [Function],
              ],
            },
            "agent": false,
            "binaryType": null,
            "ca": null,
            "cert": null,
            "ciphers": null,
            "enablesXDR": false,
            "forceBase64": false,
            "forceJSONP": false,
            "forceNode": false,
            "hostname": "localhost",
            "id": null,
            "isReactNative": false,
            "jsonp": true,
            "key": null,
            "onlyBinaryUpgrades": undefined,
            "passphrase": null,
            "path": "/socket.io/",
            "perMessageDeflate": Object {
              "threshold": 1024,
            },
            "pfx": null,
            "pingInterval": null,
            "pingIntervalTimer": null,
            "pingTimeout": null,
            "pingTimeoutTimer": null,
            "policyPort": 843,
            "port": "80",
            "prevBufferLen": 0,
            "query": Object {
              "draftId": "",
            },
            "readyState": "opening",
            "rejectUnauthorized": true,
            "rememberUpgrade": false,
            "secure": false,
            "timestampParam": "t",
            "timestampRequests": undefined,
            "transport": XHR {
              "_callbacks": Object {
                "$close": Array [
                  [Function],
                ],
                "$drain": Array [
                  [Function],
                ],
                "$error": Array [
                  [Function],
                ],
                "$packet": Array [
                  [Function],
                ],
              },
              "agent": false,
              "ca": null,
              "cert": null,
              "ciphers": null,
              "enablesXDR": false,
              "extraHeaders": undefined,
              "forceNode": false,
              "hostname": "localhost",
              "isReactNative": false,
              "key": null,
              "localAddress": undefined,
              "passphrase": null,
              "path": "/socket.io/",
              "pfx": null,
              "pollXhr": Request {
                "_callbacks": Object {
                  "$data": Array [
                    [Function],
                  ],
                  "$error": Array [
                    [Function],
                  ],
                },
                "agent": false,
                "async": true,
                "ca": null,
                "cert": null,
                "ciphers": null,
                "data": null,
                "enablesXDR": false,
                "extraHeaders": undefined,
                "index": 0,
                "isBinary": undefined,
                "key": null,
                "method": "GET",
                "passphrase": null,
                "pfx": null,
                "rejectUnauthorized": true,
                "requestTimeout": undefined,
                "supportsBinary": false,
                "uri": "http://localhost/socket.io/?draftId=&EIO=3&transport=polling&t=MbeszI0&b64=1",
                "xd": true,
                "xhr": XMLHttpRequest {
                  "DONE": 4,
                  "HEADERS_RECEIVED": 2,
                  "LOADING": 3,
                  "OPENED": 1,
                  "UNSENT": 0,
                  "abort": [Function],
                  "addEventListener": [Function],
                  "dispatchEvent": [Function],
                  "getAllResponseHeaders": [Function],
                  "getRequestHeader": [Function],
                  "getResponseHeader": [Function],
                  "handleError": [Function],
                  "onreadystatechange": [Function],
                  "open": [Function],
                  "readyState": 1,
                  "removeEventListener": [Function],
                  "responseText": "",
                  "responseXML": "",
                  "send": [Function],
                  "setDisableHeaderCheck": [Function],
                  "setRequestHeader": [Function],
                  "status": null,
                  "statusText": null,
                },
                "xs": false,
              },
              "polling": true,
              "port": "80",
              "query": Object {
                "EIO": 3,
                "b64": 1,
                "draftId": "",
                "t": "MbeszI0",
                "transport": "polling",
              },
              "readyState": "opening",
              "rejectUnauthorized": true,
              "requestTimeout": undefined,
              "secure": false,
              "socket": [Circular],
              "supportsBinary": false,
              "timestampParam": "t",
              "timestampRequests": undefined,
              "xd": true,
              "xs": false,
            },
            "transportOptions": Object {},
            "transports": Array [
              "polling",
              "websocket",
            ],
            "upgrade": true,
            "upgrades": null,
            "writeBuffer": Array [],
          },
          "lastPing": null,
          "nsps": Object {
            "/": [Circular],
          },
          "opts": Object {
            "hostname": "localhost",
            "path": "/socket.io",
            "port": "80",
            "query": Object {
              "draftId": "",
            },
            "secure": false,
          },
          "packetBuffer": Array [],
          "readyState": "opening",
          "skipReconnect": false,
          "subs": Array [
            Object {
              "destroy": [Function],
            },
            Object {
              "destroy": [Function],
            },
            Object {
              "destroy": [Function],
            },
          ],
          "uri": "http://localhost",
        },
        "json": [Circular],
        "nsp": "/",
        "query": Object {
          "draftId": "",
        },
        "receiveBuffer": Array [],
        "sendBuffer": Array [
          Object {
            "data": Array [
              "join",
              Object {
                "name": "myname1111111111111",
              },
            ],
            "id": 0,
            "options": Object {
              "compress": true,
            },
            "type": 2,
          },
        ],
        "subs": Array [
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
        ],
      }
    }
  />
</div>
`;
